x-pack/plugins/security/server/lib/authorization/check_privileges.js

```diff
@@ -38,19 +32,19 @@ export function checkPrivilegesWithRequestFactory(server) {
         }
       });
 
-      const hasPrivileges = privilegeCheck.application[application][DEFAULT_RESOURCE];
+      const privilegeCheckPrivileges = privilegeCheck.application[application][DEFAULT_RESOURCE];
```

Comment:
nit: naming here is confusing to me. If you're planning on addressing naming in a different PR, then ignore this for now.

I'm not sure if I like it any better, but for the sake of starting a discussion, what about calling `checkApplicationPrivileges` something like `checkApplicationActions`?
I know we are using the `has_privileges` API here, but in reality, we are only checking if the user is authorized to perform a particular `action`, regardless of which permission grants them that action.

Continuing with this example, then this variable could be named something like `actionCheckResults` (or `privilegeCheckResults` if we don't change the other name?)

If we decide on a rename, then this comment cascades to other functions/variables/files, but the sake of brevity, I won't call them out in this review

Reply:
These are the "new" names as recommended by: https://github.com/elastic/kibana/pull/19723#discussion_r199286813 if we would like to rename this again, I'd prefer to address it in another PR.